src/pages/costModels/costModel/updateMarkupDialog.tsx

```diff
@@ -32,23 +38,52 @@ interface Props extends WithTranslation {
 
 interface State {
   markup: string;
+  origIsDiscount: boolean;
+  isDiscount: boolean;
 }
 
 class UpdateMarkupModelBase extends React.Component<Props, State> {
   constructor(props) {
     super(props);
+    const initialMarkup = this.props.current.markup.value;
+    const isNegative = Number(initialMarkup) < 0;
+    const noSignValue = isNegative ? initialMarkup.substring(1) : initialMarkup;
+
     this.state = {
       markup:
-        (formatValue(Number(this.props.current.markup.value), 'markup', {
+        (formatValue(Number(noSignValue), 'markup', {
           fractionDigits: 2,
         }) as string) || '0.00',
+      origIsDiscount: isNegative,
+      isDiscount: isNegative,
     };
   }
+
+  private handleSignChange = (_, event) => {
+    const { value } = event.currentTarget;
+    this.setState({ isDiscount: value === 'true' });
+  };
+
+  private handleMarkupDiscountChange = (_, event) => {
+    const { value } = event.currentTarget;
+    this.setState({ markup: value });
+  };
+
+  private markupValidator = () => {
+    return /^\d*(\.?\d{1,2})?$/.test(this.state.markup) ? 'default' : 'error';
+  };
+
+  private lettersOnly = event => {
+    const regex = /[^0-9.]/g;
+    event.target.value = event.target.value.replace(regex, '');
```

Comment:
We should probably use `currentTarget` Vs `target`.

Reply:
We should consider renaming this method, since we don't accept letters (e.g., handleOnKeyDown, numbersOnly, etc.)?